transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutboxImpl.java

```diff
@@ -236,16 +236,19 @@ private <T> T schedule(Class<T> clazz, String uniqueRequestId) {
                           extracted.getMethodName(),
                           extracted.getParameters(),
                           extracted.getArgs(),
-                          uniqueRequestId);
+                          uniqueRequestId,
+                          groupId);
                   validator.validate(entry);
                   persistor.save(extracted.getTransaction(), entry);
-                  extracted
-                      .getTransaction()
-                      .addPostCommitHook(
-                          () -> {
-                            listener.scheduled(entry);
-                            submitNow(entry);
-                          });
+                  if (groupId == null) {
+                    extracted
```

Comment:
Could we change the log if `groupId  != null` to something like

```
Scheduled {} for running in next flush
```

Reply:
Are you suggesting to simply change the `after transaction commit` a bit further down to `in next flush`?

Reply:
> Are you suggesting to simply change the `after transaction commit` a bit further down to `in next flush`?

It makes sense to me to keep the existing log for `groupId == null` and use a new log for `groupId != null` - the suggested `Scheduled {} for running in next flush`.

Also, I guess `com.gruelbox.transactionoutbox.TransactionOutboxListener#scheduled` should be called for `groupId != null` when the transaction completes (without submitting).

Reply:
@badgerwithagun I see what you mean now. The log statements have now been fixed to accurately reflect the difference between unordered (run after transaction commit) and ordered (process in next flush).